src/modules/puzzle_cookware/puzzle_cookware.ino

```diff
@@ -0,0 +1,411 @@
+const int8_t seafood_tartare_ingredients[] PROGMEM = {54, 6, 9, 48};
+const int8_t sea_bass_poele_ingredients[] PROGMEM = {56, 9, 13, 22, 34};
+const int8_t sweetbread_poele_ingredients[] PROGMEM = {0, 38, 21};
+const int8_t beef_saute_ingredients[] PROGMEM = {49, 16, 13, 1};
+const int8_t beef_bourguignon_ingredients[] PROGMEM = {29, 31, 30, 18, 1};
+const int8_t stuffed_quail_ingredients[] PROGMEM = {23, 50, 21, 27, 34, 2};
+const int8_t filet_mignon_poele_ingredients[] PROGMEM = {52, 47, 24, 29, 28, 3};
+const int8_t chocolate_mousse_ingredients[] PROGMEM = {45, 38, 47};
+const int8_t fresh_fruit_compote_ingredients[] PROGMEM = {41, 43, 36};
+const int8_t crepes_suzette_ingredients[] PROGMEM = {39, 51, 47};
+const int8_t berries_au_fromage_ingredients[] PROGMEM = {44, 38, 37, 40, 42};
+const int8_t warm_banana_souffle_ingredients[] PROGMEM = {55, 47, 51};
+const int8_t fruit_gelee_ingredients[] PROGMEM = {57, 41, 37};
+const int8_t tarte_aux_fruits_ingredients[] PROGMEM = {52, 53, 54, 55, 56, 57, 58};
+
+const recipe recipes[] PROGMEM = {
+    {mushroom_terrine_ingredients, 4, 5, 0},
+    {scallop_poele_ingredients, 2, 2, 0},
+    {ratatouille_ingredients, 5, 4, 0},
+    {lobster_mousse_ingredients, 3, 5, 0},
+    {caprese_salad_ingredients, 4, 3, 0},
+    {consomme_ingredients, 3, 4, 1},
+    {pumpkin_veloute_ingredients, 2, 5, 1},
+    {carrot_potage_ingredients, 4, 3, 1},
+    {crab_bisque_ingredients, 5, 2, 1},
+    {cold_tomato_soup_ingredients, 3, 3, 1},
+    {sole_meuniere_ingredients, 2, 4, 2},
+    {eel_matelote_ingredients, 3, 2, 2},
+    {bouillabaisse_ingredients, 5, 3, 2},
+    {sea_bass_en_papillote_ingredients, 4, 3, 2},
+    {seafood_tartare_ingredients, 4, 3, 2},
+    {sea_bass_poele_ingredients, 5, 2, 2},
+    {sweetbread_poele_ingredients, 3, 2, 3},
+    {beef_saute_ingredients, 4, 3, 3},
+    {beef_bourguignon_ingredients, 5, 3, 3},
+    {stuffed_quail_ingredients, 6, 4, 3},
+    {filet_mignon_poele_ingredients, 6, 2, 3},
+    {chocolate_mousse_ingredients, 3, 5, 4},
+    {fresh_fruit_compote_ingredients, 3, 3, 4},
+    {crepes_suzette_ingredients, 3, 2, 4},
+    {berries_au_fromage_ingredients, 5, 5, 4},
+    {warm_banana_souffle_ingredients, 3, 5, 4},
+    {fruit_gelee_ingredients, 3, 4, 4},
+    {tarte_aux_fruits_ingredients, 7, 5, 4},
+};
+
+const uint8_t digits[] = {
+  7, 1, 4, 3,
+  5, 6, 0, 1,
+  4, 5, 8, 9,
+  0, 9, 2, 7,
+  3, 6, 8, 2
+};
+
+#define NUM_RECIPES 28
+#define SHOWN_INGREDIENTS 10
+
+// Global buffers
+
+char text_buffer[17];
+int8_t ingredients_buffer[7];
+
+// Global variables
+
+int8_t available_ingredients[SHOWN_INGREDIENTS];
+int8_t selected_ingredients[SHOWN_INGREDIENTS];
+int8_t ingredient_index;
+int8_t selected_recipe_index;
+recipe selected_recipe_val;
+uint32_t last_time_left;
+uint32_t millis_at_last_update;
+
+void setup() {
+  Serial.begin(115200);
+  for (int8_t i = 0; i < 6; i++) {
+    all_buttons[i].setDebounceTime(100);
```

Comment:
Lange debouncetijd gaat er misschien voor zorgen dat spelers denken dat de knop kapot is, misschien dit op 50 zetten?